front-end/src/app/reports/f3x/submission-workflow/submit-f3x-step1.component.ts

```diff
@@ -0,0 +1,184 @@
+import { schema as f3xSchema } from 'fecfile-validate/fecfile_validate_js/dist/F3X';
+import { F3xSummary } from 'app/shared/models/f3x-summary.model';
+import { F3xSummaryService } from 'app/shared/services/f3x-summary.service';
+import { CommitteeAccount } from 'app/shared/models/committee-account.model';
+import { ReportCodeLabelList } from '../../../shared/utils/reportCodeLabels.utils';
+import { updateLabelLookupAction } from '../../../store/label-lookup.actions';
+import { selectReportCodeLabelList } from 'app/store/label-lookup.selectors';
+
+@Component({
+  selector: 'app-submit-f3x-step1',
+  templateUrl: './submit-f3x-step1.component.html',
+  styleUrls: ['./submit-f3x-step1.component.scss'],
+})
+export class SubmitF3xStep1Component implements OnInit, OnDestroy {
+  formProperties: string[] = [
+    'confirmation_email_1',
+    'confirmation_email_2',
+    'change_of_address',
+    'street_1',
+    'street_2',
+    'city',
+    'state',
+    'zip',
+  ];
+  report: F3xSummary | undefined;
+  report_code = '';
+  stateOptions: PrimeOptions = [];
+  countryOptions: PrimeOptions = [];
+  formSubmitted = false;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  committeeAccount$: Observable<CommitteeAccount> = this.store.select(selectCommitteeAccount);
+  reportCodeLabelList$: Observable<ReportCodeLabelList> = new Observable<ReportCodeLabelList>();
+  form: FormGroup = this.fb.group(this.validateService.getFormGroupFields(this.formProperties));
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private f3xSummaryService: F3xSummaryService,
+    private validateService: ValidateService,
+    private fb: FormBuilder,
+    private store: Store,
+    private messageService: MessageService
+  ) {}
+
+  ngOnInit(): void {
+    // Refresh committee account details whenever page loads
+    this.store.dispatch(refreshCommitteeAccountDetailsAction());
+    this.stateOptions = LabelUtils.getPrimeOptions(StatesCodeLabels);
+    this.countryOptions = LabelUtils.getPrimeOptions(CountryCodeLabels);
+
+    this.report = this.activatedRoute.snapshot.data['report'];
+    if (this.report?.report_code) {
+      if (this.report?.report_code?.length > 0) {
+        this.report_code = this.report.report_code;
+      }
+    } else {
+      this.report_code = '';
+    }
+    this.store
+      .select(selectCommitteeAccount)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((committeeAccount) => this.setDefaultFormValues(committeeAccount));
+
+    this.reportCodeLabelList$ = this.store.select<ReportCodeLabelList>(selectReportCodeLabelList);
+    this.store.dispatch(updateLabelLookupAction());
+
+    // Initialize validation tracking of current JSON schema and form data
+    this.validateService.formValidatorSchema = f3xSchema;
+    this.validateService.formValidatorForm = this.form;
+    this.form.controls['confirmation_email_1'].addValidators(this.buildEmailValidator('confirmation_email_1'));
```

Comment:
Can this be replaced with the angular provided validators? Something like: .addValidators([Validators.required, Validators.maxLength(44), Validators.email] ? That would spare us the maintenance of the custom code and eliminate that RegEx code small. You'll still need some custom code to handle the duplicate email part.

Reply:
The Angular-provided required and maxLength validators are a great addition.

The Angular email validator can be made to work with the form as seen here:
![image](https://user-images.githubusercontent.com/104506225/180074879-581052e5-4c2e-414d-861d-6cd2feaf374e.png)

But the detection is somewhat poor
![image](https://user-images.githubusercontent.com/104506225/180075151-d48582d7-67fa-4793-b697-3bd0929f5448.png)

Might still be worth it anyways to make CloudSonar happy